### Leitura do arquivo .pkl

Leitura do arquivo onde foi salvo os previsores e as classes com seu devido pre processsamento

In [1]:
import pickle

In [2]:
with open('../../variaveis_preprocessadas/credit.pkl', 'rb') as f:
  previsores_treinamento, classe_treinamento, previsores_teste, classe_teste = pickle.load(f)

### Utilização do algoritmo Àrvores de decisão

In [4]:
from sklearn.tree import DecisionTreeClassifier

tree = DecisionTreeClassifier()
tree.fit(previsores_treinamento, classe_treinamento)

previsoes = tree.predict(previsores_teste)

### Avaliação dos resultados do algoritmo Àrvores de decisão

In [5]:
from sklearn.metrics import accuracy_score, confusion_matrix, classification_report

In [6]:
score = accuracy_score(classe_teste, previsoes)
score

0.978

In [7]:
confusion_matrix(classe_teste, previsoes)

array([[425,   7],
       [  4,  64]], dtype=int64)

In [9]:
print(classification_report(classe_teste, previsoes))

              precision    recall  f1-score   support

           0       0.99      0.98      0.99       432
           1       0.90      0.94      0.92        68

    accuracy                           0.98       500
   macro avg       0.95      0.96      0.95       500
weighted avg       0.98      0.98      0.98       500



### Resultados obtidos

O algoritmo possui 97% de acurácia. O algoritmo para a deteção de pessoas que não vão pagar o emprestimo consegue detectar 94% e possui uma acurácia de 90%. Em relação ao algoritmo de Naive Bayes, o algoritmo de árvore de decisão possui um melhor desempenho.